[winson8222] iP

winson8222 opened the pull request:
# DukePro
> Something Something in the month of May.

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _REASONABLY FAST_ to use

All you need to do is,

download it from [here](https://github.com/winson8222/ip/releases/download/A-Jar/duke.jar).
double-click it.
add your tasks.
let it manage your tasks for you 😉

And it is **FREE!**

Features:

1. Managing tasks
1. Managing deadlines (coming soon)
1. Reminders (coming soon)

This is the main Duke class in `Duke.java`
```
    public static void main(String[] args) {

        new Duke(LISTPATH).run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

winson8222 committed:
Level0

winson8222 committed:
Added echo

winson8222 committed:
Added list

winson8222 committed:
Added mark as done

winson8222 committed:
Added mark as done

winson8222 committed:
Added different task types

winson8222 committed:
Added different task types

winson8222 committed:
Added first test

winson8222 committed:
Added first test

winson8222 committed:
Added some exceptions

winson8222 committed:
Added delete

winson8222 committed:
Added Enum

winson8222 committed:
Added save

winson8222 committed:
Added Date

winson8222 committed:
added UI and Storage

winson8222 committed:
Added package

winson8222 committed:
Added package

winson8222 committed:
added parser

winson8222 committed:
added parser

winson8222 committed:
Merge tag 'A-MoreOOP2'

Added packages

winson8222 committed:
added parser to package

winson8222 committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into A-Gradle

winson8222 committed:
Added-gradle

winson8222 committed:
Merge branch 'A-Gradle'

winson8222 committed:
Added gradle build

winson8222 committed:
Merge branch 'A-Gradle'

winson8222 committed:
Added JUnit

winson8222 committed:
changed style

winson8222 committed:
Added find

winson8222 committed:
nothing

winson8222 committed:
fixed UI

winson8222 committed:
Merge branch 'branch-A-CodingStandard'

winson8222 committed:
Fixed merge conflict

winson8222 committed:
added JavaDoc

winson8222 committed:
added coding standard

winson8222 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Commands.java

winson8222 committed:
fixed conflict with javadoc

winson8222 committed:
Add GUI

tanshiyu1999 commented on src/main/java/Dates.java:
```diff
@@ -5,9 +5,11 @@
 
-    public static boolean checkDateinput(String date) {
+    static final Pattern datepattern = Pattern.compile("^(0?[1-9]|[12][0-9]|3[01])/(0?[1-9]|1[0-2])/(\\d{4}) (\\d{4}$)");
+    static final Pattern dateStringPattern = Pattern.compile("\\d{1,2} (January|February|March|April|May|June|July|August|September|October|November|December) \\d{4} \\d{2}:\\d{2}");
```
Good idea using regular expression to represent your dates!

tanshiyu1999 commented on src/main/java/UI.java:
```diff
@@ -0,0 +1,29 @@
+        }
+        System.out.println(Greeting.linebreak);
+    }
+}
```
Might be better to handle the printing of error message with your DukeException instead of printing it out with a Ui static function.

tanshiyu1999 commented on src/main/java/Greeting.java:
```diff
@@ -1,5 +1,13 @@
+ *
+ */
 public class Greeting {
     static final String linebreak = "____________________________________________________________";
```
```
suggestion
static final String LINEBREAK = "______________________"
```

tanshiyu1999 reviewed (commented):
Looks good!

Overall very well defined class system. Only nit pick was your final variable.

evanyan13 commented on src/main/java/duke/Commands.java:
```diff
@@ -0,0 +1,86 @@
+                String command = parser.getCommand();
+                CommandType given = CommandType.valueOf(command);
+
+                switch (given) {
```
Maybe add implement the change in swich-case style as mentioned in the java coding standard given to us: 
**No indentation as 'case'**  - [Tweak: switch-case style](https://se-education.org/guides/tutorials/intellijCodeStyle.html#tweak-switch-case-style)

evanyan13 commented on src/main/java/duke/Commands.java:
```diff
@@ -0,0 +1,86 @@
+                String line = sc.nextLine();
+                Parser parser = new Parser(line);
+                String command = parser.getCommand();
+                CommandType given = CommandType.valueOf(command);
```
Maybe can be more specific here in terms of variable naming, instead of just `given`, can put `commandGiven`?
```suggestion
                CommandType commandGiven = CommandType.valueOf(command);
```

evanyan13 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,31 @@
+ * Duke is entry point of our program
+ */
+public class Duke {
+    public static final String LISTPATH = "./data/list.txt";
```
I like how you extracted this out as a final variable instead of hardcoding each line with the file path

evanyan13 commented on src/main/java/duke/Greeting.java:
```diff
@@ -0,0 +1,28 @@
+public class Greeting {
+
+    /**
+     * This method prints out the line when user start the program
```
In method header comments, the first sentence can start in the form `Returns ...`, `Sends ...`, `Adds ...`.
So I guess we need not to say `This method...` because it is already the method's header comment?

evanyan13 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,102 @@
+            if (line.charAt(0) == 'T') {
+                String[] para = line.split(" | ");
+                String description = para[4];
+                ToDo newtask = new ToDo(description);
```
Maybe we can use this format for variable/instance
`newTask`
Applies to some other variables in this file as well

Actually, it we already know it is a ToDo event, why not we just do this instead:
```suggestion
                ToDo todo = new ToDo(description);
```

evanyan13 reviewed (commented):
LGTM 👍  Just that some variable naming may not be camelCase like `newtask` / `newitem`

winson8222 committed:
Add assertion statements

The assertion statements to check if save file exist, so that it can be loaded.
It also check if /fxml file exists to ensure JavaFX will function properly.

It prevents situation where the files are accidentally moved or deleted.

winson8222 committed:
change parser

winson8222 committed:
Improve Code Quality

Add some of the repeated, pattern and string patterns with no explanation into final fields on class.

winson8222 committed:
Add reschedule function

User can now reschedule deadline and event to another time.

winson8222 committed:
Merge pull request #1 from winson8222/branch-B-Snooze

Branch b snooze

winson8222 committed:
Improve Code Quality

Change some repeated Strings into final variable and rename some variable
to follow standards.

winson8222 committed:
Merge pull request #2 from winson8222/branch-A-CodeQuality

Improve Code Quality

winson8222 committed:
Merge branch 'master' into branch-A-Assertions

winson8222 committed:
Merge pull request #3 from winson8222/branch-A-Assertions

Add assertion statements

winson8222 committed:
Change styling

Rearrange the order of the static varibles in Parser.

Lines that are too long are modified to follow coding standard.

winson8222 committed:
Merge branch 'master' of https://github.com/winson8222/ip
Some styling is fixed

winson8222 committed:
Abstract logics to reduce method length

winson8222 committed:
Update .gitignore to ignore .class files

Commands.class, CommandType.class, Commands.class are ignored.

winson8222 committed:
Change README

README now includes a user guide to show basic functions of Chat Box.

Basic commands that users can used are included in README.

winson8222 committed:
Adds JUnit test cases

winson8222 committed:
Update user guide

winson8222 committed:
Rename PNG file

winson8222 committed:
Add exit

winson8222 committed:
Add file not exist handling

winson8222 committed:
Add file not exist handling

winson8222 committed:
Update README
